js/oceanex.js

```diff
@@ -0,0 +1,323 @@
+                'base': base,
+                'quote': quote,
+                'baseId': base,
+                'quoteId': quote,
+                'active': true,
+                'info': market,
+                'precision': undefined,
+                'limits': undefined,
+            });
+        }
+        // parse markets
+        return result;
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        let response = await this.publicGetTickersPair (this.extend ({ 'pair': symbol }), params);
+        const data = response['data'];
+        return this.parseTicker (data, symbol);
+    }
+
+    parseTicker (data, market = undefined) {
+        let timestamp = data['at'];
+        const ticker = data['ticker'];
+        if (typeof timestamp === 'string') {
+            if (timestamp.length > 0) {
+                timestamp = this.parse8601 (timestamp);
+            }
+        }
+        return {
+            'symbol': market,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeFloat (ticker, 'high'),
+            'low': this.safeFloat (ticker, 'low'),
+            'bid': this.safeFloat (ticker, 'buy'),
+            'bidVolume': undefined,
+            'ask': this.safeFloat (ticker, 'sell'),
+            'vwap': undefined,
+            'open': undefined,
+            'close': this.safeFloat (ticker, 'last'),
+            'last': this.safeFloat (ticker, 'last'),
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeFloat (ticker, 'volume'),
+            'quoteVolume': undefined,
+            'info': ticker,
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        let response = await this.publicGetOrderBook (this.extend ({
+            'market': symbol,
+            'limit': limit,
+        }, params));
+        const orderbook = this.safeValue (response, 'data');
+        return this.parseOrderBook (orderbook, orderbook['timestamp']);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        let response = await this.publicGetTrades (this.extend ({
+            'market': symbol,
+            'limit': limit,
+        }, params));
+        let data = this.safeValue (response, 'data');
+        return this.parseTrades (data, symbol, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        let side = undefined;
+        if (trade['side'] === 'bid') {
+            side = 'buy';
+        } else if (trade['side'] === 'ask') {
+            side = 'sell';
+        }
+        return {
+            'info': trade,
+            'timestamp': trade['created_on'],
+            'datetime': trade['created_at'],
+            'symbol': trade['market'],
+            'id': trade['id'],
+            'order': undefined,
+            'type': 'limit',
```

Comment:
If the exchange supports `limit` orders only, then we need to designate `has: { 'createMarketOrder': false }` in [`descibe()` above](https://github.com/ccxt/ccxt/pull/5091/files#diff-27adc3166fcc2d328d8e1ac942896cb0R62).